### PR TITLE
CONTRIBUTING.md: Typo fixes, https

### PR DESCRIPTION
**Pull request by Margen67**
### Description of Changes
Fix typos and make PCSX2 forum link https.

### Rationale behind Changes
See description.

### Suggested Testing Steps
Read it.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # So you want to contribute to PCSX2? Great
 
-As a first step, please review these links as they'll help you understand how the development of the PCSX2 works. These pages are all found on the github [wiki](https://github.com/PCSX2/pcsx2/wiki). You should probably also check out the [forums](http://forums.pcsx2.net/index.php).
+As a first step, please review these links as they'll help you understand how the development of PCSX2 works. These pages are all found on the GitHub [wiki](https://github.com/PCSX2/pcsx2/wiki). You should probably also check out the [forums](https://forums.pcsx2.net/index.php).
```
Comment (GovanifY) [inline .github/CONTRIBUTING.md:3]: @xTVaser @RedDevilus Do we really want to direct people to forums in the contribution guide?

**Review by GovanifY — COMMENTED** [review]
LGTM, thanks! The comment is unrelated to your PR, but still a good idea to talk about it now

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (RedDevilus) [reply .github/CONTRIBUTING.md:3]: I don't mind personally due to good threads for people.

Comment (GovanifY) [reply .github/CONTRIBUTING.md:3]: Which threads? Care to give an example of something that would actually help new contributors?
Not against it but I'm not sure which usefulness new devs can get out of it

Comment (RedDevilus) [reply .github/CONTRIBUTING.md:3]: I suppose it's more patch creators and compatability reports and some other threads i can't think of. I agree it's more a sidenote than an obvious pro to contributors.

Comment (GovanifY) [reply .github/CONTRIBUTING.md:3]: I'll wait for @xTVaser to give his thoughts and maybe reformulate this then

Comment (xTVaser) [reply .github/CONTRIBUTING.md:3]: I don't have a strong opinion (I've never been on the forums myself).  But imo, if you are contributing the forums isn't really the relevant place at all.  If the wiki has critical docs related to developing, then that should be pulled out and centralized where the development actually happens -- github.

**Pull request merged**